### PR TITLE
feat: passwordless SMS and expiry notice in code / link templates

### PR DESCRIPTION
**Pull request by aeneasr**
This feature allows Ory Kratos to use the SMS gateway for login and registration with code via SMS.

Additionally, the default email and sms templates have been updated. We now also expose `ExpiresInMinutes` / `expires_in_minutes` in the templates, making it easier to remind the user how long the code or link is valid for.

## Related issue(s)

Closes https://github.com/ory/kratos/issues/1570
Closes https://github.com/ory/kratos/issues/3779

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [x] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got the approval (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](https://github.com/ory/docs).

## Further Comments

* [ ] Add e2e tests

### CODE REVIEW

File: text/message_registration.go [hunk 1]
```diff
@@ -84,7 +84,7 @@ func NewRegistrationEmailWithCodeSent() *Message {
 	return &Message{
 		ID:   InfoSelfServiceRegistrationEmailWithCodeSent,
 		Type: Info,
-		Text: "An email containing a code has been sent to the email address you provided. If you have not received an email, check the spelling of the address and retry the registration.",
+		Text: "A code has been sent to the address(es) you provided. If you have not received an message, check the spelling of the address and retry the registration.",
```
Comment (jonas-jonas) [inline text/message_registration.go:87]:
```suggestion
		Text: "A code has been sent to the address you provided. If you have not received a message, check the spelling of the address and retry the registration.",
```

When could this be more than one address?

Comment (aeneasr) [reply text/message_registration.go:87]: Registration where both email and phone number are given and both are used for otp login

Comment (jonas-jonas) [reply text/message_registration.go:87]: Makes sense.

Just a typo then :) 

```suggestion
		Text: "A code has been sent to the address(es) you provided. If you have not received a message, check the spelling of the address and retry the registration.",
```

**Comment by eddmann** [comment]
Hey 👋🏻 , thanks for all the great work you do on Ory.

I am very interested in this feature, providing the ability to support a passwordless login based on an SMS OTP. Just wondering would the flow for this authentication method be for the user to provide their phone number and then based on that identifier being present in the database it will send an SMS code to the phone number? Similar to how the Email OTP works?

**Comment by aeneasr** [comment]
The feature behaves the same, it's just supporting another notificatino channel now.

**Comment by eddmann** [comment]
Great, apologies for another question. To expand upon this will the user supply their email address or phone number and based on that send the login code to the relevant notification channel?

For example, sending the following request would send an email notification:

```
<input type="hidden" name="identifier" value={emailAddress} />
```

And sending the following request would send an SMS notification, providing the phone number is an identifier?

```
<input type="hidden" name="identifier" value={phoneNumber} />
```

or can we supply the email address as the identifier but send an SMS OTP instead, perhaps with an additional parameter?

```
<input type="hidden" name="channel" value="phone|email" />
```

**Comment by vinckr** [comment]
Hello @eddmann 

generally we can't give out ETAs for new OSS releases - right now the cadence is 2-3 times per year, depending on other work and priorities. 

If you rely on Ory Kratos as a business critical component I would suggest to look into purchasing an Ory Enterprise License - this will give you access to the same binary that is used for the [managed service](https://console.ory.sh/) where a release is cut for every bugfix/new feature; so you always get access to the latest improvements as they are merged. 
Plus it comes with 24/7 production support, SLAs, compliance, etc. Feel free to [contact the team](https://ory.sh/contact) or reach out to me on our [community slack](https://slack.ory.sh/) directly if you are interested.

**Pull request merged**